platform/ios/src/MGLMapView.mm

```diff
@@ -4733,6 +4757,53 @@ - (void)updateAnnotationViews
     [CATransaction commit];
 }
 
+- (void)updateCalloutView
+{
+    UIView <MGLCalloutView> *calloutView = self.calloutViewForSelectedAnnotation;
+    id <MGLAnnotation> annotation = calloutView.representedObject;
+    
+    if (calloutView && annotation)
+    {
+        MGLAnnotationTag tag = [self annotationTagForAnnotation:annotation];
+        MGLAnnotationContext &annotationContext = _annotationContextsByAnnotationTag.at(tag);
+        MGLAnnotationView *annotationView = annotationContext.annotationView;
+        
+        // Update callout view for view annotation
+        if (annotationView)
+        {
+            CGPoint point = annotationView.center;
+            point.x += annotationView.centerOffset.dx;
+            point.y += annotationView.centerOffset.dy;
+            point.y -= CGRectGetMidY(annotationView.bounds);
```

Comment:
The center (and thus the bounds and frame) already account for the center offset. Sorry for being unclear about that previously. This does mean you can combine the view and non-view cases, just like you do in the `MapChangeRegionWillChange` handler above.